.local/bin/kblayoutmanager

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
+# An adaptation of https://github.com/porras/i3-keyboard-layout/blob/master/i3-keyboard-layout by MYDavoodeh
+set -e
+
+get_kbdlayout() { setxkbmap -query | grep -oP 'layout:\s*\K(\w+)' ;}
+
+set_kbdlayout() {
+	# FIXME what to do for variants?
+	setxkbmap "$1"
+	pgrep i3status | xargs --no-run-if-empty kill -s USR1 # to tell i3status to update
+}
+
+cycle() { #TODO: Optimize cycling time
+	current_layout=$(get_kbdlayout)
+	eval last="\$$#"
+	# If current_layout is the last one OR current_layout is not in the given layouts, set the first one in cycle as the initiator. Else find the next.
+	{ [ "$last" = "$current_layout" ] || ! echo "$*" | grep -qo "$current_layout" ;} && set_kbdlayout "$1" || set_kbdlayout "$(echo "$*" | cut -d' ' -f "$#")"
```

Comment:
Nice addition, but it only works with two or fewer layouts defined.
`set_kbdlayout "$(echo "$*" | cut -d' ' -f "$#")"` always selects the last element in list.
It could be fixed with something like this:
`set_kbdlayout $(echo "$*" | sed -E "s/^.*$current_layout ([a-zA-Z]*) .*$/\1/")`

Reply:
Thanks for your comment.
Indeed there is a problem... It's funny I haven't noticed it in years...

I'm afraid I couldn't make your suggested solution to work... (I'm just a novice sorry)

But I'll work on it tomorrow in order to fix it...
Sorry for bad English

Reply:
This works for me, I had an extra space in my original comment :smiley: 
```suggestion
	{ [ "$last" = "$current_layout" ] || ! echo "$*" | grep -qo "$current_layout" ;} && set_kbdlayout "$1" || set_kbdlayout $(echo "$*" | sed -E "s/^.*$current_layout ([a-zA-Z]*).*$/\1/")
```

Reply:
Thanks a lot...
I've tested it, and it's pretty robust!

I'll edit my files and post them in another issue... Seems like I've deleted my fork a long time ago...